File: app/lib/models/viagem.dart
class Viagem {
  const Viagem({
    required this.id,
    required this.destino,
    required this.dataIda,
    required this.dataVolta,
    required this.finalidade,
    required this.transporte,
    this.observacoes,
    required this.status,
    required this.criadoEm,
  });

  final int id;
  final String destino;
  final DateTime dataIda;
  final DateTime dataVolta;
  final String finalidade;
  final String transporte;
  final String? observacoes;
  final String status;
  final DateTime criadoEm;

  factory Viagem.fromJson(Map<String, dynamic> json) {
    return Viagem(
      id: json['id'] as int,
      destino: json['destino'] as String,
      dataIda: DateTime.parse(json['dataIda'] as String),
      dataVolta: DateTime.parse(json['dataVolta'] as String),
      finalidade: json['finalidade'] as String,
      transporte: json['transporte'] as String,
      observacoes: json['observacoes'] as String?,
      status: json['status'] as String,
      criadoEm: DateTime.parse(json['criadoEm'] as String),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'destino': destino,
      'dataIda': dataIda.toIso8601String(),
      'dataVolta': dataVolta.toIso8601String(),
      'finalidade': finalidade,
      'transporte': transporte,
      'observacoes': observacoes,
      'status': status,
      'criadoEm': criadoEm.toIso8601String(),
    };
  }
}

File: viagens-backend/src/main/java/com/agt/viagens/domain/model/Viagem.java
package com.agt.viagens.domain.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.ZoneOffset;

@Entity
@Table(
    name = "viagens",
    indexes = {
        @Index(name = "idx_viagem_usuario_criado", columnList = "usuario_id, criado_em DESC")
    }
)
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Viagem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 200)
    private String destino;

    @Column(name = "data_ida", nullable = false)
    private LocalDate dataIda;

    @Column(name = "data_volta", nullable = false)
    private LocalDate dataVolta;

    @Column(nullable = false, length = 100)
    private String finalidade;

    @Column(nullable = false, length = 100)
    private String transporte;

    @Column(length = 1000)
    private String observacoes;

    @Builder.Default
    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private StatusViagem status = StatusViagem.AGENDADA;

    @Column(name = "criado_em", nullable = false, updatable = false)
    private LocalDateTime criadoEm;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "usuario_id", nullable = false)
    private Usuario usuario;

    @PrePersist
    void prePersist() {
        if (this.criadoEm == null) this.criadoEm = LocalDateTime.now(ZoneOffset.UTC);
    }
}
